src/engine/source/builder/test/src/unit/policy/factory_test.cpp

```diff
@@ -464,6 +464,11 @@ INSTANTIATE_TEST_SUITE_P(
             AT::DECODER, "decoder/parent1", "output/asset")(AT::DECODER, "decoder/parent2", "decoder/parent3")(
             AT::DECODER, "decoder/parent3", "decoder/Input")(AT::RULE, "rule/asset", "rule/Input")(
             AT::OUTPUT, "output/asset", "output/Input"))),
+        // Decoders forming cycles (self-loop, two-node, multi-node)
```

Comment:
besides the specific scenario it should be great to Unit Test each of the new methods